rust/agents/relayer/src/merkle_tree/builder.rs

```diff
@@ -81,54 +78,43 @@ impl MerkleTreeBuilder {
     #[instrument(err, skip(self), level="debug", fields(prover_latest_index=self.count()-1))]
     pub fn get_proof(
         &self,
-        leaf_index: u32,
+        message_nonce: u32,
         root_index: u32,
-    ) -> Result<Proof, MerkleTreeBuilderError> {
+    ) -> Result<Option<Proof>, MerkleTreeBuilderError> {
+        let Some(message_id) = self
```

Comment:
could probably do some maps instead of these but not a big deal

Reply:
problem is that if you `map`/`and_then` you end up with nested `Option<Result<Option<...>>>`. Managed to skip one let-else though

Reply:
Found a way to neatly chain everything and removed let-else entirely 🙆‍♂️